crates/sui-core/src/transaction_manager.rs

```diff
@@ -559,12 +553,6 @@ impl TransactionManager {
                     "Ignoring enqueued certificate from wrong epoch. Expected={} Certificate={:?}",
                     inner.epoch, pending_cert.certificate
                 );
-                // also ensure the transaction will not be retried after restart.
```

Comment:
Are you sure we can get rid of all these removals?
For instance, say state sync is ahead of consensus on a validator, so you could have transactions show up in state sync and checkpoint executor first, then latter on they show up in consensus which will still be added to pending_execution. These will never be removed because we are already past the checkpoint executor for them.

Reply:
You're right, good point. But is it worth fixing it, given that this is a temporary workaround? The table will be deleted at the end of the epoch anyway, and the extra transactions won't hurt anything if the node restarts again.